vmas/simulator/environment/environment.py

```diff
@@ -770,6 +773,61 @@ def render(
         # render to display or array
         return self.viewer.render(return_rgb_array=mode == "rgb_array")
 
+    def plot_boundary(self):
+
+        # include boundaries in the rendering
+        from vmas.simulator.rendering import Line
+        from vmas.simulator.utils import Color
+
+        # check if the environment is dimension-limited
+        if self.world.x_semidim is not None or self.world.y_semidim is not None:
+            # set a big value for the cases where the environment is dimension-limited only in one coordinate
+            infinite_value = 100
+
+            x_semi = self.world.x_semidim if self.world.x_semidim else infinite_value
+            y_semi = self.world.y_semidim if self.world.y_semidim else infinite_value
+
+            # set the color for the boundary line
+            color = Color.GRAY.value
+
+            # Define boundary points based on whether world semidims are provided
+            if self.world.x_semidim is not None and self.world.y_semidim is not None:
+                boundary_points = [
+                    (-x_semi, y_semi),
+                    (x_semi, y_semi),
+                    (x_semi, -y_semi),
+                    (-x_semi, -y_semi),
+                ]
+            elif self.world.x_semidim is not None:
+                boundary_points = [
+                    (-x_semi, y_semi),
+                    (-x_semi, -y_semi),
+                    (x_semi, y_semi),
+                    (x_semi, -y_semi),
+                ]
+            elif self.world.y_semidim is not None:
+                boundary_points = [
+                    (-x_semi, y_semi),
+                    (x_semi, y_semi),
+                    (x_semi, -y_semi),
+                    (-x_semi, -y_semi),
+                ]
```

Comment:
this looks the same as the first part of the if

Reply:
I have just fixed